ganalytics.cpp

```diff
@@ -416,6 +424,27 @@ void GAnalytics::Private::readMessagesFromFile(const QList<QString> &dataList)
 }
 
 /**
+ * Change the user id.
+ * @param userID         A string with the user id.
+ */
+void GAnalytics::Private::setUserID(QString userID)
+{
+    this->userID = userID;
+    QSettings settings;
+    settings.setValue("GAnalytics-uid", userID);
+}
+
+/**
+ * Get the user id.
+ * User id once created is stored in application settings.
+ * @return userID         A string with the user id.
+ */
+QString GAnalytics::Private::getUserID()
+{
+    return userID;
```

Comment:
Your setUserID sets the QSettings value but you never read it. getUserID is called from your constructor, so it is an empty string in any way?